src/dafny/Relations.dfy

```diff
@@ -52,4 +52,13 @@ module {:options "-functionSyntax:4"} Dafny.Relations {
     && Connected(R)
   }
 
+  /** An element in an ordered set is called minimal, if it is less than every other element of the set. */
+  ghost predicate IsMinimum<T>(R: (T, T) -> bool, m: T, s: set<T>) {
```

Comment:
Could you give a more descriptive name to the relation, like `lessThanOrEquals` ?

Reply:
Note that all predicates in `Relations.dfy` are currently using the name `R`. In my opinion, using `lessThanOrEquals` instead is misleading (and too long), as we are talking about _generic_ relations, not the particular case where the relation is given by the familiar `<=` relation (on e.g. the naturals), which satisfies specific properties.

Reply:
Sounds good :)

Reply:
Just some extra context:

My hope was that using the name `lessThanOrEquals` would _only_ indicate that IsMinimum is true if the relation always holds when `m` is on the _left_ of the relation, not anything else like whether the relation is totally or partially ordered. Without that name, how is a user supposed to know when to use `IsMinimum` vs `IsMaximum`?

If I look at the Wiki definition of `minimum` for a generic relation, it also uses the terms "greater than or equal to", without implying anything about partially vs totally ordered.